Added use-cases (#27)
kubescape/kubescape.io: docs/use_cases/argocd-gates-using-kubescape.md

#### **Scenario**

> A DevOps/Platform team is using ArgoCD for Kubernetes orchestration. They seek to embed security into the workflow to ensure every deployment is secure and compliant with industry benchmarks. To achieve this, Kubescape is chosen to implement security gates within their ArgoCD pipelines. Thus, automatically preventing delivery of any non-compliant deployments.

Reviewer: I think this scenario section is great, my only request here is to make it less specific on the user side (instead of "DevOps/Platform team is using ArgoCD..." use wording like "When users are using ArgoCD for Kubernetes orchestration and they seek to embed security into..."

> A DevOps/Platform team is using ArgoCD for Kubernetes orchestration. They seek to embed security into the workflow to ensure every deployment is secure and compliant with industry benchmarks. To achieve this, Kubescape is chosen to implement security gates within their ArgoCD pipelines. Thus, automatically preventing delivery of any non-compliant deployments.

Reviewer: I think this scenario section is great, my only request here is to make it less specific on the user side (instead of "DevOps/Platform team is using ArgoCD..." use wording like "When users are using ArgoCD for Kubernetes orchestration and they seek to embed security into..."

> To reinforce security within their software delivery workflow, a team is leveraging ArgoCD for Kubernetes orchestration. They seek to ensure every deployment is secure and compliant with industry benchmarks. To achieve this, Kubescape is chosen to be integrated within their ArgoCD pipelines as a security gate, automatically preventing any non-compliant deployments.

Author: ```suggestion
A DevOps/Platform team is using ArgoCD for Kubernetes orchestration. They seek to embed security into the workflow to ensure every deployment is secure and compliant with industry benchmarks. To achieve this, Kubescape is chosen to implement security gates within their ArgoCD pipelines. Thus, automatically preventing delivery of any non-compliant deployments.
```

> To reinforce security within their software delivery workflow, a team is leveraging ArgoCD for Kubernetes orchestration. They seek to ensure every deployment is secure and compliant with industry benchmarks. To achieve this, Kubescape is chosen to be integrated within their ArgoCD pipelines as a security gate, automatically preventing any non-compliant deployments.

Author: Fixed

## **Actions**

### Kubescape Integration in ArgoCD Workflows:

> Kubescape is woven into the ArgoCD deployment process, with its scans initiated through additional steps in the Application resource definition.

Author: ```suggestion
Kubescape is woven into the ArgoCD deployment process, with its scans initiated through adding additional steps in the Application resource definition.
```

Sync hooks invoke Kubescape as a last check, ensuring no security issues are introduced during the pipeline's execution.

### Policy Enforcement with Kubescape:

The team curates a selection of scan policies for Kubescape, adopting standards like CIS Benchmarks, NIST guidelines, or tailored organizational security protocols.

### Deployment Interruption on Detection of Risks:

Deployments are configured to be halted upon discovery of any critical issues by Kubescape, preventing progression of non-compliant changes.

### Security Issue Review and Resolution:

If Kubescape's scans block a deployment, the team examines the scan reports to pinpoint and resolve security concerns, guided by Kubescape’s remediation advice.

### Policy and Threshold Optimization:

Kubescape's policy settings and thresholds for failing a build are adjusted to align with the team's security objectives and delivery expectations.

## **Expected Outcome**

### Incorporating Kubescape within ArgoCD yields:

Automatic and comprehensive security assessments during deployments.

Assurance that only vetted, secure, and compliant applications are deployed.

> Early detection of potential security risks, facilitating prompt resolution.

Author: ```suggestion
Build-time detection of potential security risks, facilitating prompt resolution.
```

Mitigation of the likelihood of deploying insecure code, which enhances the organization's defense against security infractions and regulatory non-compliance.

## **Additional Benefits**

> Ingrains a preemptive security approach by embedding checks early in the CI/CD process.

Author: ```suggestion
instills a preemptive security approach by embedding checks early in the CI/CD process.
```

> Ingrains a preemptive security approach by embedding checks early in the CI/CD process.

Author: ```suggestion
Instills a preemptive security approach by embedding checks early in the CI/CD process.
```
